src/worker/sandbox/cgroups/pool.go

```diff
@@ -23,20 +22,52 @@ type CgroupPool struct {
 	recycled chan *CgroupImpl
 	quit     chan chan bool
 	nextID   int
+	logger   slog.Logger
 }
 
 func NewCgroupPool(name string) (*CgroupPool, error) {
+
+	level := new(slog.LevelVar)
```

Comment:
Won't this kind of logic be needed in lots of places?  Best to do at the "common" level